federation/src/main/scala/caliban/federation/tracing/ApolloFederatedTracing.scala

```diff
@@ -98,41 +98,50 @@ object ApolloFederatedTracing {
     wrapPureValues: Boolean
   ): FieldWrapper[Any] =
     new FieldWrapper[Any](wrapPureValues) {
+
+      private def updateState(startTime: Long, fieldInfo: FieldInfo)(result: Either[ExecutionError, ResponseValue]) =
+        ZQuery.succeed {
```

Comment:
Do we even need to wrap this? We only execute it within a foldQuery. Could we just execute purely and then lift the result?

Reply:
Yeah I think that's better actually. I'll do the change